rpc/src/v1/impls/net.rs

```diff
@@ -22,25 +22,31 @@ use v1::traits::Net;
 
 /// Net rpc implementation.
 pub struct NetClient<S: ?Sized> {
-	sync: Arc<S>
+	sync: Arc<S>,
+	/// Cached `network_id`.
+	///
+	/// We cache it to avoid redundant aquire of sync read lock.
+	/// https://github.com/paritytech/parity-ethereum/issues/8746
+	network_id: u64,
 }
 
 impl<S: ?Sized> NetClient<S> where S: SyncProvider {
 	/// Creates new NetClient.
 	pub fn new(sync: &Arc<S>) -> Self {
 		NetClient {
 			sync: sync.clone(),
+			network_id: sync.status().network_id,
 		}
 	}
 }
 
 impl<S: ?Sized> Net for NetClient<S> where S: SyncProvider + 'static {
 	fn version(&self) -> Result<String> {
-		Ok(format!("{}", self.sync.status().network_id).to_owned())
+		Ok(format!("{}", self.network_id))
 	}
 
 	fn peer_count(&self) -> Result<String> {
-		Ok(format!("0x{:x}", self.sync.status().num_peers as u64).to_owned())
+		Ok(format!("0x{:x}", self.sync.status().num_peers as u64))
```

Comment:
use {:#x} instead of 0x{:x}

Reply:
This is probably also pretty slow, we could maybe exposes a specific method to return only number of peers (best would be if it didnt require locking at all) - but it's a separate issue.

Reply:
@tomusdrw definitely, but let's fix it in a separate pr :)

Reply:
@tomusdrw 
Just so I understand this, do you mean to return like ```u64``` instead of a String here to get rid of allocation including probably something like ```itoa``` under the hood to convert the integer to ascii/unicode?

Reply:
I'm pretty sure @tomusdrw was talking about accessing `sync.status()` object which aquires lock :)

Reply:
^^ that @niklasad1 :)